### PR TITLE
iOS-2595 Search | New global search module with backlinks logic

### PR DESCRIPTION
**Pull request by joe-pusya**
https://github.com/anyproto/anytype-swift/assets/6065952/16b509fd-6ec4-499a-bd98-cbe32f129147

### CODE REVIEW

File: Anytype/Sources/PresentationLayer/Flows/HomeFlow/HomeCoordinatorViewModel.swift [hunk 1]
```diff
@@ -233,14 +234,25 @@ final class HomeCoordinatorViewModel: ObservableObject,
...
-                AnytypeAnalytics.instance().logSearchResult()
-                self?.openObject(screenData: data.editorScreenData)
-            }
-        )
+        
+        if FeatureFlags.newGlobalSearch {
+            showGlobalSearchData = GlobalSearchModuleData(
+                spaceId: activeWorkspaceStorage.workspaceInfo.accountSpaceId,
+                onSelect: { [weak self] screenData in
+                    AnytypeAnalytics.instance().logSearchResult()
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/Flows/HomeFlow/HomeCoordinatorViewModel.swift:242]: I think you can move analytics to module (logScreenSearch and logSearchResult)

File: Anytype/Sources/ServiceLayer/Object/Search/SearchService.swift [hunk 2]
```diff
@@ -62,14 +62,15 @@ final class SearchService: SearchServiceProtocol {
         return try await searchMiddleService.search(filters: filters, sorts: [sort], fullText: "", limit: Constants.defaultLimit)
     }
     
-    func search(text: String, limitObjectIds: [String]) async throws -> [ObjectDetails] {
+    func search(text: String, limitObjectIds: [String], spaceId: String) async throws -> [ObjectDetails] {
```
[the remark this answers is not in the document]
Comment (mgolovko) [reply Anytype/Sources/ServiceLayer/Object/Search/SearchService.swift:65]: objectId - is unique in network. limitObjectIds make spaces limitation for search

[the remark this answers is not in the document]
Comment (mgolovko) [reply Anytype/Sources/ServiceLayer/Object/Search/SearchService.swift:65]: If limitObjectIds contains in different spaces, spaceId will make a additional limit

**Review by mgolovko — APPROVED** [review]

File: Anytype/Sources/ServiceLayer/Object/Search/SearchService.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (joe-pusya) [reply Anytype/Sources/ServiceLayer/Object/Search/SearchService.swift]: got it, removed

File: Anytype/Sources/PresentationLayer/Flows/HomeFlow/HomeCoordinatorViewModel.swift (hunk 1 above)
Comment (joe-pusya) [reply Anytype/Sources/PresentationLayer/Flows/HomeFlow/HomeCoordinatorViewModel.swift]: Moved

**Pull request merged**